### ISSUE TITLE
Update to Angular 10

### ISSUE THREAD

**Issue by Nightsphere** [OpenWaterFoundation/owf-app-infomapper-ng#283]
Even though the InfoMapper has recently been updated from Angular 7 to 9, version 10 has now been out for long enough that updating would be a good idea. Other major packages such as Angular Material have also been upgraded to 10, and will work nicely with the new Angular version; there are already some newly implemented features that I'm missing out on and can add to the InfoMapper once upgraded.

**Comment by Nightsphere** [comment]
The changes for the update to Angular 10 is now merged. The Angular update guide helped with a lot of the process, and most
of the changes were done automatically using some `ng` commands. The biggest complaints I got were:

1. Some issues with third-party npm packages that use Angular as a dependency. They all just needed to be updated to their respective latest stable releases which had support for Angular 10. Therefore, a `npm i` will need to be performed to update them on the next git pull.
2. There was a warning in the console about using the common JS package `showdown`. In the `angular.json` file, I added a new `allowedCommonJsDependency` property:

   ```json
   "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "allowedCommonJsDependencies": [
              "showdown"
            ]
         }
      }
   ```
Number 2 is more of a reminder for me to add to the developer documentation later, as it was the only thing not covered in the update guide. I'll leave this issue open for now in case an overlooked update issue shows up.
